libcudacxx/test/libcudacxx/cuda/bit/bitfield.pass.cpp

```diff
@@ -0,0 +1,79 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the libcu++ Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#include <cuda/bit>
+#include <cuda/std/cassert>
+#include <cuda/std/cstdint>
+#include <cuda/std/type_traits>
+
+#include "test_macros.h"
+
+template <typename T>
+__host__ __device__ constexpr bool test()
+{
+  using nl             = cuda::std::numeric_limits<T>;
+  constexpr T all_ones = static_cast<T>(~T{0});
+  unused(all_ones);
+  assert(cuda::bitfield_insert(T{0}, all_ones, 0, 1) == 1);
+  assert(cuda::bitfield_insert(T{0}, all_ones, 1, 1) == 0b10);
+  assert(cuda::bitfield_insert(T{0b10}, all_ones, 0, 1) == 0b11);
+  assert(cuda::bitfield_insert(all_ones, all_ones, 0, 1) == all_ones);
+  assert(cuda::bitfield_insert(all_ones, all_ones, 2, 1) == all_ones);
+
+  assert(cuda::bitfield_insert(T{0}, all_ones, 0, 2) == 0b11);
+  assert(cuda::bitfield_insert(T{0}, all_ones, 3, 2) == 0b11000);
+  assert(cuda::bitfield_insert(T{0b10100000}, all_ones, 3, 2) == 0b10111000);
+  assert(cuda::bitfield_insert(T{0b10100000}, T{0b11}, 3, 2) == 0b10111000);
+  assert(cuda::bitfield_insert(T{0}, all_ones, nl::digits - 1, 1) == (T{1} << (nl::digits - 1u)));
+  assert(cuda::bitfield_insert(T{0b10100000}, all_ones, 0, nl::digits) == all_ones);
```

Comment:
We should also add a test case that overwrites the some set bits in the input. Currently, all tests could also be covered with bit-wise `or`'ing